force_wfmanager/left_side_pane/new_entity_modal.py

```diff
@@ -0,0 +1,106 @@
+from traits.api import (HasStrictTraits, Instance, List, Button, Either,
+                        on_trait_change, Dict)
+from traitsui.api import (View, Handler, HSplit, VGroup, UItem,
+                          HGroup, ListStrEditor, InstanceEditor)
+from traitsui.list_str_adapter import ListStrAdapter
+
+from force_bdss.api import (BaseMCOModel, BaseMultiCriteriaOptimizerBundle,
+                            BaseDataSourceModel, BaseDataSourceBundle,
+                            BaseKPICalculatorModel, BaseKPICalculatorBundle)
+
+from .workflow_model_view import WorkflowModelView
+from .view_utils import get_bundle_name
+
+
+class ListAdapter(ListStrAdapter):
+    """ Adapter for the list of available MCOs/Data sources/KPI calculators
+    bundles """
+    def get_text(self, object, trait, index):
+        return get_bundle_name(self.item)
+
+
+class ModalHandler(Handler):
+    def object_add_button_changed(self, info):
+        if info.object.current_model is not None:
+            info.object.workflow.add_entity(info.object.current_model)
+        info.ui.dispose(True)
+        info.object.clear_model()
+
+    def object_cancel_button_changed(self, info):
+        info.ui.dispose(False)
+        info.object.clear_model()
+
+
+class NewEntityModal(HasStrictTraits):
+    """ Dialog which allows the user to add a new MCO/Data Source/KPI
+    calculator to the workflow """
+    workflow = Instance(WorkflowModelView)
+
+    #: Available bundles
```

Comment:
This dialog is generic and can contain any bundle depending on the right menu cluck original source.